modules/tensor_mechanics/src/actions/TensorMechanicsAxisymmetricRZAction.C

```diff
@@ -35,24 +36,27 @@ void
 TensorMechanicsAxisymmetricRZAction::act()
 {
   std::vector<NonlinearVariableName> displacements = getParam<std::vector<NonlinearVariableName> > ("displacements");
+  unsigned int _ndisp = displacements.size();
   std::vector<VariableName> coupled_displacements;
-  unsigned int dim = displacements.size();
 
-  //Error checking:  Can only take two displacement variables in AxisymmetricRZ
-  mooseAssert(dim == 2, "Expected two displacement variables but recieved " << dim);
+  //Error checking:  Can only take two displacement variables in StressDivergenceRZTensors kernel
+  mooseAssert(_ndisp == 2, "Expected two displacement variables but recieved " << _ndisp);
 
-  for (unsigned int i = 0; i < dim; ++i)
+  for (unsigned int i = 0; i < _ndisp; ++i)
     coupled_displacements.push_back(displacements[i]);
 
-  // Must retain the size two vector here so the materials know which variable is where
-  std::vector<std::vector<AuxVariableName> > save_in(2);
+  std::vector<std::vector<AuxVariableName> > save_in(_ndisp);
   if (isParamValid("save_in_disp_r"))
     save_in[0] = getParam<std::vector<AuxVariableName> >("save_in_disp_r");
-
   if (isParamValid("save_in_disp_z"))
     save_in[1] = getParam<std::vector<AuxVariableName> >("save_in_disp_z");
 
-  // Set up the information needed to pass to create the new kernel
+  std::vector<std::vector<AuxVariableName> > diag_save_in(_ndisp);
+  if (isParamValid("diag_save_in_disp_r"))
+    diag_save_in[0] = getParam<std::vector<AuxVariableName> >("diag_save_in_disp_r");
+  if (isParamValid("diag_save_in_disp_z"))
+    diag_save_in[1] = getParam<std::vector<AuxVariableName> >("diag_save_in_disp_z");
```

Comment:
All this logic is obsolete in the vector coupling case. This is how we should go forward!